FirebaseAuth/Sources/Swift/AuthProvider/PhoneAuthProvider.swift

```diff
@@ -525,7 +525,7 @@ import Foundation
         )
       }
 
-      return try await withCheckedThrowingContinuation { continuation in
+      return try await withUnsafeThrowingContinuation { continuation in
```

Comment:
Interested in discussing ideas for transforming this to a task-based approach.

Reply:
I wonder if this had something to do with the method called on line 529. It dispatches to the main queue so maybe there was a runtime violation it picked up.